examples/inspection/plot_partial_dependence.py

```diff
@@ -163,179 +299,221 @@
+}
+_, ax = plt.subplots(ncols=3, figsize=(10, 4))
 tic = time()
 display = PartialDependenceDisplay.from_estimator(
-    est_no_interactions,
+    hgbdt_model_without_interactions,
     X_train,
-    ["MedInc", "AveOccup", "HouseAge", "AveRooms"],
-    kind="both",
-    subsample=50,
-    n_jobs=3,
-    grid_resolution=20,
-    random_state=0,
-    ice_lines_kw={"color": "tab:blue", "alpha": 0.2, "linewidth": 0.5},
-    pd_line_kw={"color": "tab:orange", "linestyle": "--"},
+    **features_info,
+    ax=ax,
+    **common_params,
 )
-
 print(f"done in {time() - tic:.3f}s")
-display.figure_.suptitle(
-    "Partial dependence of house value with Gradient Boosting\n"
-    "and no interactions allowed"
+_ = display.figure_.suptitle(
+    "1-way vs 2-way of numerical PDP using gradient boosting", fontsize=16
 )
-display.figure_.subplots_adjust(wspace=0.4, hspace=0.3)
+plt.subplots_adjust(wspace=0.3)
 
 # %%
-# All 4 plots have parallel ICE lines meaning there is no interaction in the
+# The 1D partial dependence plots for the model constrained to not model feature
+# interactions show local spikes for each features individually, in particular for
+# for the "humidity" feature. Those spikes might be reflecting a degraded behavior
+# of the model that attempts to somehow compensate for the forbidden interactions
+# by overfitting particular training points. Note that the predictive performance
+# of this model as measured on the test set is significantly worse than that of
+# the original, unconstrained model.
+#
+# Also note that the number of local spikes visible on those plots is depends on
+# the grid resolution parameter of the PD plot itself.
+#
+# Those local spikes result in a noisily gridded 2D PD plot. It is quite
+# challenging to tell whether or not there are no interaction between those
+# features because of the high frequency oscillations in the humidity feature.
+# However it can clearly be seen that the simple interaction effect observed when
+# the temperature crosses the 20 degrees boundary is no longer visible for this
 # model.
-# Let us also have a look at the corresponding 2D-plot.
-
+#
+# The partial dependence between categorical features will provide a discrete
+# representation that can be shown as a heatmap. For instance the interaction between
+# the season, the weather, and the target would be as follow:
 print("Computing partial dependence plots...")
+features_info = {
+    "features": ["season", "weather", ("season", "weather")],
+    "kind": "average",
+    "categorical_features": categorical_features,
+}
+_, ax = plt.subplots(ncols=3, figsize=(14, 6), constrained_layout=True)
 tic = time()
-_, ax = plt.subplots(ncols=3, figsize=(9, 4))
 display = PartialDependenceDisplay.from_estimator(
-    est_no_interactions,
+    hgbdt_model,
     X_train,
-    ["AveOccup", "HouseAge", ("AveOccup", "HouseAge")],
-    kind="average",
-    n_jobs=3,
-    grid_resolution=20,
+    **features_info,
     ax=ax,
+    **common_params,
 )
```

Comment:
For some reason, this figure has a truncated `ylim` but @jeremiedbb and I do not understand why. This problem does not happen to the other plots above. Changing `constrained_layout` does not seem to impact this.

`pdp_lim` is left to its default value.